spec/wrap.md

```diff
@@ -0,0 +1,104 @@
+# WRAP Specification
+
+WRAP defines the rules for bi-directional communication between a Host and a Wasm module by defining the processes of invocation and subinvocation. It makes use of Host allocated Wasm shared memory to pass encoded buffers between the Host and the Wasm module.
+
+The Wasm module must define the following imports and exports:
+
+#### Exports
+- `_invoke: (bufferLen: u32) => u32`
```

Comment:
Should we rename "bufferLen" to "argsLen"? I think this buffer-use-case-specific naming is helpful for readers.

Reply:
Renamed to `optBufferLen` to match the convention we have in the rest of the document.
It's the invoke options buffer length not the args buffer. Args buffer is inside of the invoke options.